core/remotes/docker/fetcher.go

```diff
@@ -262,7 +264,27 @@ func (r dockerFetcher) FetchByDigest(ctx context.Context, dgst digest.Digest, op
 	return seeker, desc, nil
 }
 
+const (
+	_        = iota
+	kB int64 = 1 << (10 * iota)
+	mB
+	gB
+	tB
+	pB
+	eB
+
+	// When under this size, a layer will not be fetched using parallel requets
+	minimumSizeForParallelDL = 1 * mB
+)
+
 func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string, offset int64) (_ io.ReadCloser, retErr error) {
```

Comment:
Can you help modularize this function, its already somewhat big enough to break it down.

Reply:
yes !

Reply:
Hm, I tried a bit, and in this case I think breaking it down would only complicate things and make it harder to understand, etc.

Do you confirm you'd really like me to split it up ?